File: go/api/Syslog.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Syslog 对应 Syslog.py。
type Syslog struct {
	*session.ApiGroup
}

// NewSyslog 创建 Syslog API 分组。
func NewSyslog(s *session.Session) *Syslog {
	return &Syslog{ApiGroup: session.NewApiGroup(s)}
}

// Querylog 对应 syslog/querylog。
func (s *Syslog) Querylog() (map[string]interface{}, error) {
	return s.S.Get("syslog/querylog", nil, "api")
}

// Clear 清空系统日志。对应 syslog/processlog (post_set, {"command":"clear"})。
func (s *Syslog) Clear() (interface{}, error) {
	return s.S.PostSet("syslog/processlog", map[string]interface{}{
		"command": "clear",
	}, false, "api", false, false)
}

File: src/api/syslog.rs
//! Syslog API group (`api/Syslog.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;
use crate::tools::map_of;
use crate::xml::XmlMap;

use super::{get_value, post_set};

/// Syslog API group.
pub struct Syslog<'a> {
    conn: &'a Connection,
}

impl<'a> Syslog<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Syslog { conn }
    }

    /// `syslog/querylog`. Query the syslog.
    pub fn querylog(&self) -> Result<Value> {
        get_value(self.conn, "api/syslog/querylog")
    }

    /// Clear the syslog (process-log `command=clear`).
    pub fn clear(&self) -> Result<String> {
        let body: XmlMap = map_of([("command", "clear".to_string())]);
        post_set(self.conn, "api/syslog/processlog", &body)
    }
}
